### PR TITLE
Fix task-state availability and production diagnostics

### PR DESCRIPTION
**Pull request by sambitcreate**
## Summary

Fix the confirmed task-state and production-diagnostics defects found while investigating the desktop “Tasks unavailable” banner.

### Separate fixes

1. **Authoritative task replay** — validate the newest successful full task checkpoint, allowing it to supersede invalid ancestors while still rejecting an invalid newest checkpoint.
2. **Honest durable tracking** — share task-state loading between chat-open and generation, omit the durable task tool for journalless turns, distinguish storage-disabled tracking from invalid snapshots, and validate tool output before mutation/publication.
3. **Rollout refresh and cohort preservation** — reread device policy, serialize concurrent loads, publish complete initial JSON atomically, reject invalid/deleted replacements, and preserve the new-chat activation cutoff as stages widen.
4. **Privacy-preserving diagnostics** — retain bounded structural causes/statuses, distinguish renderer exceptions from process crashes, count cancellations correctly, classify provider errors before outcome redaction, and record distinct task/MCP evidence without raw text.

### Review process

Each fix received three independent `opencode-go/deepseek-v4.1-flash` adversarial reviews. Rollout and diagnostics received additional three-review rounds after fixing edge cases found during review. Review lenses covered correctness/integration, hostile inputs/races/privacy, and behavioral coverage.

### Verification

- `npm test`
- `npm run test:todo`
- `npm run test:compaction`
- `npm run test:diagnostics`
- Focused provider-failure and harness diagnostic tests
- `npm run type-check`
- `npm run lint`
- `npm run build`
- `npm run test:e2e:diagnostics:production`

Desktop todo IPC has no iOS/Android consumers. Portable provider-failure and shared-server contracts remain compatible.

### Scope of the evidence

The old production logs discarded the causes of some provider, renderer, MCP and shutdown errors, so those historical causes remain unknown. This PR improves future diagnostic evidence. Production rollout advancement is a receipt-validated operator action; storage-disabled chats gain durable tracking when their cohort is enabled. These source changes reach the installed app through the normal release process.

### CODE REVIEW

File: main/services/pi-upgrade-rollout.ts [hunk 1]
```diff
@@ -355,9 +366,14 @@ export class PiUpgradeRolloutStore {
...
-      const next = { version: 1 as const, stage: target, activatedAt: this.now(), revision: current.revision + 1 };
+      const next = {
+        version: 1 as const, stage: target,
+        // Establish the new-chat cohort once; later stages only widen it.
+        activatedAt: target === "new_chats" ? this.now() : current.activatedAt,
+        revision: current.revision + 1,
+      };
       await atomicPrivateJson(paths.root, paths.policy, next);
-      this.loaded = Promise.resolve(next);
+      this.hasLoaded = true;
```
[the remark this answers is not in the document]
Comment (sambitcreate) [reply main/services/pi-upgrade-rollout.ts:377]: Fixed in fc96b21c. Every validated read now latches policy observation, including reads made by a subsequently failing advance. The ENOENT initialization branch rechecks that latch after asynchronous work. Added a deterministic regression with an overlapping first load, failed advance, and policy deletion. Three independent DeepSeek V4.1 Flash reviews passed; compaction suite and full npm test passed.

**Comment by sambitcreate** [comment]
Addressed both actionable Pullfrog observations: fc96b21c closes failed-advance policy observation, and 6f3bf778 preserves main-generated RD-UUIDv4 references through journal normalization with an exact structural grammar. Each follow-up received three fresh DeepSeek V4.1 Flash reviews (30 reviews total); full npm test, diagnostics/compaction suites, typecheck, lint and production diagnostics E2E passed. The hard-link requirement is an accepted atomic-publication tradeoff consistent with existing repository storage patterns. Please recheck the updated head.

**Comment by sambitcreate** [comment]
Final verification on 6f3bf778: verify, Deterministic Electron E2E, changed-area detection and Pullfrog all succeeded; Android was correctly skipped. Pullfrog’s incremental review reports no new issues and its rollout thread is resolved. Both review follow-ups are separate commits, and 30 independent DeepSeek V4.1 Flash adversarial reviews are complete. Working tree clean; PR ready for merge.
